core/src/main/java/org/apache/gravitino/authorization/UserGroupManager.java

```diff
@@ -109,6 +113,25 @@ User getUser(String metalake, String user) throws NoSuchUserException {
     }
   }
 
+  String[] listUserNames(String metalake) {
+    return Arrays.stream(listUsers(metalake)).map(User::name).toArray(String[]::new);
+  }
+
+  User[] listUsers(String metalake) {
+    try {
+      Namespace namespace = AuthorizationUtils.ofUserNamespace(metalake);
+      return store.list(namespace, UserEntity.class, Entity.EntityType.USER).stream()
+          .map(entity -> (User) entity)
```

Comment:
Why do we need this line?

Reply:
So you only get the user list here, right? Will you also fetch the roles for each user, I saw that `User` and `UserDTO` have `roles` fields, will this be empty?

Reply:
Yes, they are empty. Only they are used, we fetch them.

Reply:
You can see the pull request. https://github.com/apache/gravitino/pull/4690
UserEntity `roles` field don't be fetched directly. We will put a supplier method field instead. So we don't fetch the values directly if we only need the names.
If we need all the information of entities, we can also have the ability to put them. You can see the list details. Because the front side need the all the users information, so I provide the interface.

Reply:
I mean here why do we need to do a type conversion `entity -> (User) entity`?

Reply:
Sorry, it's unnecessary. I ever designed other interface `listEntitiesByRelation`. It need me to use type cast. For this place, we don't  need. I have removed it.